lib/components/user/places/favorite-places-pane.js

```diff
@@ -1,60 +1,49 @@
-import { FieldArray } from 'formik'
-import React, { Component } from 'react'
+import React from 'react'
 import { ControlLabel } from 'react-bootstrap'
+import { connect } from 'react-redux'
 
-import { isHomeOrWork } from '../../../util/user'
+import * as userActions from '../../../actions/user'
+import { PLACES_PATH } from '../../../util/constants'
+import { cloneUserDataForEditing, isHomeOrWork } from '../../../util/user'
 import FavoritePlaceRow from './favorite-place-row'
 
 /**
  * Renders an editable list user's favorite locations, and lets the user add a new one.
+ * Additions, edits, and deletions of places take effect immediately.
  */
-class FavoriteLocationsPane extends Component {
-  _handleDelete = (arrayHelpers, place, index) =>
-    () => {
-      const isFixed = isHomeOrWork(place)
-      if (isFixed) {
-        // Just clear the address if place is fixed.
-        const newPlace = {
-          ...place,
-          address: ''
-        }
-        arrayHelpers.replace(index, newPlace)
-      } else {
-        arrayHelpers.remove(index)
-      }
-    }
-
-  render () {
-    const { values: userData } = this.props
-    const { savedLocations } = userData
-
-    return (
-      <div>
-        <ControlLabel>Add the places you frequent often to save time planning trips:</ControlLabel>
-
-        <FieldArray
-          name='savedLocations'
-          render={arrayHelpers => (
-            <>
-              {savedLocations.map((place, index) => (
-                <FavoritePlaceRow
-                  isFixed={isHomeOrWork(place)}
-                  key={index}
-                  onDelete={this._handleDelete(arrayHelpers, place, index)}
-                  path={`/places/${index}`}
-                  place={place}
-                />
-              )
-              )}
-
-              {/* For adding a location. */}
-              <FavoritePlaceRow path='/places/new' />
-            </>
-          )}
+const FavoritePlacesPane = ({ deleteUserPlace, loggedInUser }) => {
```

Comment:
I'm not a huge fan of the Pane name. How about FavoritePlacesList?

Reply:
Updated in https://github.com/opentripplanner/otp-react-redux/pull/280/commits/436b93da0ef77b9c75628f92d29cf8313085d163.